doc/rm/c_cpp_coding_style.md

```diff
@@ -162,6 +162,18 @@ Any nonstandard features that are used must be compatible with both GCC and Clan
 
 This rule deviates from the Google C++ style guide to align closer with a typical way of writing C code.
 
+The names of enumeration constants must be prefixed with the name of their respective enumeration type.
+This is because C does not have namespaces, and this is clearest way to avoid name clashes.
+The exact capitalisation does not need to match, as enumeration type names have a different capitalisation rule to enumeration constants.
+For example:
+```c
+enum my_wonderful_option_t {
```

Comment:
```c
typedef enum my_wonderful_option {} my_wonderful_option_t;
```
seems to be the norm? IMO we should be writing
```
typedef enum {} my_wonderful_option_t;
```
but that seems like a discussion for another time.

Reply:
Ah yes, the `_t` is only for types in the main type namespace. will update.